be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/MessageDecoder.scala

```diff
@@ -0,0 +1,19 @@
+package ch.epfl.pop.pubsub.graph
+
+import akka.NotUsed
+import akka.http.scaladsl.model.ws.{Message, TextMessage}
+import akka.stream.scaladsl.Flow
+import ch.epfl.pop.json.JsonMessageParser
+
+object MessageDecoder {
+
+  // Message or TextMessage
+  val jsonRpcParser: Flow[Message, Nothing, NotUsed] = Flow[Message].map {
+    case TextMessage.Strict(text) => JsonMessageParser.parseMessage(text) match {
+      case _ => ???
+    }
+    case TextMessage.Streamed(stream) => ??? // throw
```

Comment:
what's this?
you mentioned something about strict/streamed last night, but I was missing some context.

Reply:
AFAIK, `TextMessage.Strict` sends (or receives) a **full** message on/from a websocket. 

The graph is able to start processing the streamed version before the whole message was received (probably for large amount of data). I don't know yet if it will be useful for us

Reply:
I would say not.
I know a bunch of protocols that are very well adapted to this kind of "streaming" but our naive JSON-RPC approach isn't one of them. Let's not go down that path.